plugin/impl.go

```diff
@@ -9,107 +15,123 @@ import (
 
 	"github.com/drone/drone-go/drone"
 	"github.com/joho/godotenv"
+	"github.com/urfave/cli/v2"
 	"golang.org/x/oauth2"
 )
 
-// Plugin defines the Downstream plugin parameters.
-type Plugin struct {
-	Repos          []string
+// Settings for the plugin.
+type Settings struct {
+	Repos          cli.StringSlice
 	Server         string
 	Host           string
 	Proto          string
 	Token          string
 	Wait           bool
 	Timeout        time.Duration
 	LastSuccessful bool
-	Params         []string
-	ParamsEnv      []string
+	Params         cli.StringSlice
+	ParamsEnv      cli.StringSlice
 	Deploy         string
+
+	server string
+	params map[string]string
 }
 
-// Exec runs the plugin
-func (p *Plugin) Exec() error {
-	if len(p.Token) == 0 {
-		return fmt.Errorf("Error: you must provide your Drone access token.")
+// Validate handles the settings validation of the plugin.
+func (p *Plugin) Validate() error {
+	if len(p.settings.Token) == 0 {
+		return fmt.Errorf("you must provide your drone access token")
 	}
 
-	p.Server = getServerWithDefaults(p.Server, p.Host, p.Proto)
-	if len(p.Server) == 0 {
-		return fmt.Errorf("Error: you must provide your Drone server.")
+	p.settings.server = getServerWithDefaults(p.settings.Server, p.settings.Host, p.settings.Proto)
+	if len(p.settings.server) == 0 {
```

Comment:
This check doesn't make sense, if no server is provided the p.settings.Server variable will be anyway empty.